docs/docs/adding-search-with-algolia.md

```diff
@@ -2,7 +2,604 @@
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+Two things before we begin:
+
+1. Beyond this guide, you may also want to checkout Algolia's extensive [docs on how to get started in React](https://www.algolia.com/doc/guides/building-search-ui/getting-started/react).
+2. If you're looking to add search to a documentation site, you can let Algolia handle most of the steps outlined below for you by using their excellent [Docsearch](https://community.algolia.com/docsearch) functionality. For other types of sites and more fine-grained control over exactly what data should be indexed, read on.
+
+## Backend
+
+First, you'll need to add [`gatsby-plugin-algolia`](https://github.com/algolia/gatsby-plugin-algolia), [`react-instantsearch-dom`](https://github.com/algolia/react-instantsearch) and [`algoliasearch`](https://github.com/algolia/algoliasearch-client-javascript) to your project. With `react-instantsearch` Algolia provides an extensive library of off-the-shelf React components that we can import to save ourselves a lot of work. `algoliasearch` provides the actual search client which we'll simply pass into `react-instantsearch`. If you're not using it already, also install [`dotenv`](https://github.com/motdotla/dotenv) while you're at it. We're going to need it to specify your Algolia app ID and both the search and admin API keys without commiting them to version control.
+
+```sh
+yarn add gatsby-plugin-algolia react-instantsearch-dom algoliasearch dotenv
+```
+
+Next, add `gatsby-plugin-algolia` to your `gatsby-config.js`.
+
+```js
+// gatsby-config.js
+const queries = require('./src/utils/algolia')
+
+require('dotenv').config()
+
+module.exports = {
+  siteMetadata: {
+    title: `Gatsby+Algolia`,
+    description: `How to setup Algolia search in Gatsby`,
+    author: <your name>,
+  },
+  plugins: [
+    {
+      resolve: `gatsby-plugin-algolia`,
+      options: {
+        appId: process.env.GATSBY_ALGOLIA_APP_ID,
+        apiKey: process.env.ALGOLIA_ADMIN_KEY,
+        queries,
+        chunkSize: 10000, // default: 1000
+      },
+    },
+  ],
+}
+```
+
+Notice that we're loading `queries` from a file at `./src/utils/algolia.js` (you can of course put it wherever you like) and our Algolia ID and API key from `.env` so let's add those files.
+
+```env
+// .env
+GATSBY_ALGOLIA_APP_ID = KA4OJA9KAS
+GATSBY_ALGOLIA_SEARCH_KEY=lkjas987ef923ohli9asj213k12n59ad
+ALGOLIA_ADMIN_KEY = lksa09sadkj1230asd09dfvj12309ajl
+```
+
+I inserted random character sequences here but yours should be the same length. Also, it's good practice to commit a `.env.example` to version control so that if someone forks your repo, they know which environment variables they need to supply.
+
+```env
```

Comment:
You can add file names with some special Markdown syntax to make this a little easier.

    ```env:title=.env.example

Reply:
I added a title to all code blocks.